.github/workflows/publish-docker.yml

```diff
@@ -11,29 +11,41 @@ on:
 
 jobs:
   publish-docker-image:
-    if: github.event_name == 'release'
+    if: ${{ github.event_name == 'release' || github.event.act }}
     runs-on: ubuntu-latest
     steps:
       - name: Checkout
         uses: actions/checkout@v3
+
       - name: Set up QEMU
         uses: docker/setup-qemu-action@v2
+
       - name: Set up Docker Buildx
         uses: docker/setup-buildx-action@v2
+
       - name: Login to Docker Hub
         uses: docker/login-action@v2
         with:
           username: ${{ secrets.DOCKERHUB_USERNAME }}
           password: ${{ secrets.DOCKERHUB_TOKEN }}
-      - name: Docker meta
-        id: meta
-        uses: docker/metadata-action@v4
+
+      - name: Build and test
+        uses: docker/build-push-action@v4
```

Comment:
Why two times the build and push action?

Reply:
yes. The first is to build, then we run the test tuite and the second is to push it once the test suite has passed. The building should be cached between the two actions. See example here https://docs.docker.com/build/ci/github-actions/test-before-push/

Reply:
Ah so if you do not add push in the `with` options, it just builds, good to know